js/previewplugin.js

```diff
@@ -73,14 +74,29 @@
 			// if a filelist is present, the PDF viewer can be closed to go back there
 			$('#pdframe').load(function(){
 				var iframe = $('#pdframe').contents();
-				if ($('#fileList').length) {
-					iframe.find('#secondaryToolbarClose').click(function() {
+				if ($('#fileList').length)
+				{
+					iframe.find('#secondaryToolbarClose').click(function() 
+					{
 						if(!$('html').hasClass('ie8')) {
 							history.back();
 						} else {
 							self.hide();
 						}
 					});
+
+					// Go back on ESC
+					$(document).keyup(function(e) {
+					     if (shown && e.keyCode == 27) // escape key maps to keycode `27`
+					     { 
+					     	shown = false;
+					        if(!$('html').hasClass('ie8')) {
+								history.back();
```

Comment:
Intendation is wrong. But since IE8 isn't supported anymore we could just say self.hide() (also in the block above).